src/common/commonutils/FileUtils.c

```diff
@@ -932,6 +949,8 @@ int ReplaceMarkedLinesInFile(const char* fileName, const char* marker, const cha
         {
             if (NULL != (tempHandle = fopen(tempFileName, "w")))
             {
+                RestrictFileAccessToCurrentAccountOnly(tempFileName);

```

Comment:
This is racy - the file is first created with potentially wide open permissions and later restricted, so there is a window where it can be manipulated.

Reply:
Please see last iteration. The file is now emptied after being access restricted, before writing to it.